employee_service/controller/employee.go

```diff
@@ -0,0 +1,198 @@
+package employee_controller
+
+import (
+	"HR-system/employee_service/models"
+	employee_storage "HR-system/employee_service/storage"
+	responses "HR-system/employee_service/utils"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+type Server struct {
+	Storage *employee_storage.Storage
+	Router  *gin.Engine
+}
+
+func (s *Server) Run(
+	host, port, user, dbname, password, sslmode string,
+) {
+	s.initializeDB(host, port, user, dbname, password, sslmode)
+	s.startServe(":8080")
+}
+
+func (server *Server) startServe(addr string) {
+	log.Println("Listening to port 8080")
+	log.Fatal(http.ListenAndServe(addr, server.Router))
+}
+
+func (s *Server) initializeDB(host, port, user, dbname, password, sslmode string) {
+	databaseUrl := fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=%s", host, port, user, dbname, password, sslmode)
+	db, err := gorm.Open(postgres.Open(databaseUrl), &gorm.Config{})
+
+	if err != nil {
+		log.Printf("Cannot connect to %s database", dbname)
+		log.Fatal("This is the error:", err)
+		return
+	}
+
+	s.Storage.SetDB(db)
+	err = s.Storage.AutoMigrate()
+	if err != nil {
+		log.Printf("Cannot migrate %s database", dbname)
+		return
+	}
+
+	s.Router = gin.Default()
+
+	s.Router.LoadHTMLGlob("templates/*")
+
+	v1 := s.Router.Group("/api/v1/employee")
+	{
+		v1.GET("/", s.index)
+		v1.GET("/all", s.getAll)
+		v1.GET("/:id", s.get)
+		v1.POST("/", s.create)
+		v1.PUT("/:id", s.update)
+		v1.DELETE("/:id", s.delete)
+	}
+}
+
+type ID struct {
+	ID uint `json:"id" uri:"id"`
+}
+
+func (s *Server) create(c *gin.Context) {
+	var employee models.EmployeeDTO
+
+	if err := c.ShouldBindJSON(&employee); err != nil {
+		responses.BadRequest(c, []string{err.Error()})
+		return
+	}
+
+	employee.SetRequired(true)
+	messages := employee.Validate()
+	if len(messages) > 0 {
+		responses.BadRequest(c, messages)
+		return
+	}
+
+	createEmployee := employee.MapForCreate()
+
+	// validate data
+
+	tx := s.Storage.Create(&createEmployee)
+	if tx.Error != nil {
+		messages = []string{tx.Error.Error()}
+		responses.BadRequest(c, messages)
```

Comment:
This should be an internal server error.

Reply:
I will fix next PR